gcp_variant_transforms/transforms/limit_write_test.py

```diff
@@ -0,0 +1,101 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for limit_write module."""
+
+from __future__ import absolute_import
+
+import unittest
+
+from apache_beam.testing.test_pipeline import TestPipeline
+from apache_beam.testing.util import assert_that
+from apache_beam.testing.util import equal_to
+from apache_beam.transforms import Create
+
+from gcp_variant_transforms.beam_io import vcfio
+from gcp_variant_transforms.transforms import limit_write
+from gcp_variant_transforms.transforms.variant_to_bigquery import _WRITE_SHARDS_LIMIT as DefaultShardLimit
+
+
+class LimitWriteTest(unittest.TestCase):
+  """Test cases for the ``LimitWrite`` PTransform."""
+
+  def _get_sample_variants(self):
+    variant1 = vcfio.Variant(
+        reference_name='chr19', start=11, end=12, reference_bases='C',
+        alternate_bases=['A', 'TT'], names=['rs1', 'rs2'], quality=2,
+        filters=['PASS'],
+        info={'IFR': vcfio.VariantInfo([0.1, 0.2], 'A'),
+              'IFR2': vcfio.VariantInfo([0.2, 0.3], 'A'),
+              'IS': vcfio.VariantInfo('some data', '1'),
+              'ISR': vcfio.VariantInfo(['data1', 'data2'], '2')},
+        calls=[
+            vcfio.VariantCall(
+                name='Sample1', genotype=[0, 1], phaseset='*',
+                info={'GQ': 20, 'FIR': [10, 20]}),
+            vcfio.VariantCall(
+                name='Sample2', genotype=[1, 0],
+                info={'GQ': 10, 'FB': True}),
+        ]
+    )
+    variant2 = vcfio.Variant(
+        reference_name='20', start=123, end=125, reference_bases='CT',
+        alternate_bases=[], filters=['q10', 's10'],
+        info={'II': vcfio.VariantInfo(1234, '1')})
+    variant3 = vcfio.Variant(
+        reference_name='20', start=None, end=None, reference_bases=None)
+    variant4 = vcfio.Variant(
+        reference_name='20', start=123, end=125, reference_bases='CT',
+        alternate_bases=[], filters=['q10', 's10'],
+        info={'II': vcfio.VariantInfo(1234, '1')},
+        calls=[
+            vcfio.VariantCall(
+                name='EmptySample', genotype=[], phaseset='*',
+                info={}),
+        ])
+    return [variant1, variant2, variant3, variant4]
+
+
+  def test_limit_write_default_shard_limit(self):
+    variants = self._get_sample_variants()
+    input_pcoll = Create(variants)
+    pipeline = TestPipeline(blocking=True)
+    output_pcoll = (
+        pipeline
+        | input_pcoll
+        | 'LimitWrite' >> limit_write.LimitWrite(DefaultShardLimit))
+    assert_that(output_pcoll, equal_to(variants))
+    pipeline.run()
+
+  def test_limit_write_shard_limit_4(self):
+    variants = self._get_sample_variants()
+    input_pcoll = Create(variants)
+    pipeline = TestPipeline(blocking=True)
```

Comment:
just curious why you need to set `blocking=true` as it was not needed in previous tests.

Reply:
I copied from merge_variants_test.py without paying attention to details :)